### PR TITLE
feat: magic wand

### PR DESCRIPTION
**Pull request by tuhacrt**
## Overview

1. [chore: update antd from 5.11 to 5.22.3](https://github.com/flux3dp/beam-studio-core/commit/b552c8dbb7fa751bcafffb8b5578168492236ee3)
2. [chore: install konva, react-konva, use-image](https://github.com/flux3dp/beam-studio-core/commit/3b9e88981537287ff371cb3c42636cb550726be2)
3. add image edit panel (eraser, magic wand) [[clickup]](https://app.clickup.com/t/21v923c)
4. revise shortcuts, use key instead of keyCode

## Related PRs

1. [[app]](https://github.com/flux3dp/beam-studio/pull/336)
2. [[web]](https://github.com/flux3dp/beam-studio-web/pull/217)

### CODE REVIEW

File: src/web/app/components/ImageEditPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+  useEffect(() => {
+    const initialize = async () => {
+      requestAnimationFrame(() => {});
+
+      const {
+        blobUrl,
+        originalWidth: width,
+        originalHeight: height,
+      } = await preprocessByUrl(src, { isFullResolution: true });
+      const fullColorImage = await calculateBase64(blobUrl, true, 255, true);
```
[the remark this answers is not in the document]
Comment (tuhacrt) [reply src/web/app/components/ImageEditPanel/index.tsx:336]: maybe pull out these calculation to workers on other PRs.

[the remark this answers is not in the document]
Comment (dkdkwizard) [reply src/web/app/components/ImageEditPanel/index.tsx:336]: Sure

**Review by dkdkwizard — APPROVED** [review]

**Review by dkdkwizard — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
     "@react-three/fiber": "^8.15.12",
```
Comment (dkdkwizard) [inline package.json:23]: 123 test

File: package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
...
     "@sentry/react": "^6.4.1",
     "@szhsin/react-menu": "^1.10.1",
-    "antd": "5.11",
+    "antd": "^5.22.3",
```
Comment (dkdkwizard) [inline package.json:26]: test 456

**Pull request merged**